src/components/crownBoundaryEventDropdown.vue

```diff
@@ -47,14 +51,20 @@ export default {
     };
   },
   methods: {
+    canAddBoundaryEventToTarget,
     addBoundaryEvent(nodeType) {
       this.dropdownOpen = false;
 
+      if (!canAddBoundaryEventToTarget(nodeType, this.shape)) {
+        return;
+      }
+
       const definition = this.nodeRegistry[nodeType].definition(this.moddle, this.$t);
       const diagram = this.nodeRegistry[nodeType].diagram(this.moddle);
+      const emptyPort = getEmptyBoundaryEventPositionsForShape(this.shape)[0];
 
-      diagram.bounds.x = this.node.diagram.bounds.x;
-      diagram.bounds.y = this.node.diagram.bounds.y;
+      diagram.bounds.x = emptyPort.x - (diagram.bounds.width / 2);
+      diagram.bounds.y = emptyPort.y - (diagram.bounds.height / 2);
```

Comment:
I had to place the _centre_ of the element onto the port. That is why I'm dividing width/height by 2 here.